File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.cs
public class Solution {
    public int[] Intersect(int[] nums1, int[] nums2) {
        //  https://leetcode.com/problems/intersection-of-two-arrays-ii/solutions/2992743/c-simple-solution
        Dictionary<int, int> d = new Dictionary<int, int>();
        foreach(int elem in nums1){
            if(d.ContainsKey(elem)){
                d[elem]++;
            }else{
                d.Add(elem,1);
            }
        }
        List<int> list = new List<int>();
        foreach(int elem in nums2){
            if(d.ContainsKey(elem)){
                list.Add(elem);
                d[elem]--;
                if(d[elem] == 0){
                    d.Remove(elem);
                }
            }
        }
        int[] n = new int[list.Count];
        int i = 0;
        foreach(int elem in list){
            n[i] = elem;
            i++;
        }
        return n;
    }
}

File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        //  https://leetcode.com/problems/intersection-of-two-arrays-ii/solutions/3099498/c-6ms-10mb-short-and-simple-using-search-and-erase
        //  원래 정렬, 중복제거 후 했던방법 대로 하려 했으나
        //  다른 방법이 궁금하여 다른 솔루션 찾아봤음
        vector<int> ans;
        //  nums1은 돌면서
        for(auto & p : nums1){
            //  nums1의 p가 nums2에 있는지 찾아보고
            auto k = find(nums2.begin(), nums2.end(), p);
            //  k!= nums2.end() 인 경우 
            //  즉, nums2에 p와 같은 수가 있으면
            if(k != nums2.end()){
                //  nums2에서 k(=p)를 지우고
                nums2.erase(k);
                //  vector에 p 삽입
                ans.push_back(p);
            }
        }
        return ans;

    }
};